server/src/main/java/org/elasticsearch/search/vectors/RescoreVectorBuilder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.search.vectors;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class RescoreVectorBuilder implements Writeable, ToXContentObject {
+
+    public static final ParseField NUM_CANDIDATES_FACTOR_FIELD = new ParseField("num_candidates_factor");
+    public static final float MIN_OVERSAMPLE = 1.0F;
+    private static final ConstructingObjectParser<RescoreVectorBuilder, Void> PARSER = new ConstructingObjectParser<>(
+        "rescore_vector",
+        args -> new RescoreVectorBuilder((Float) args[0])
+    );
+
+    static {
+        PARSER.declareFloat(ConstructingObjectParser.constructorArg(), NUM_CANDIDATES_FACTOR_FIELD);
+    }
+
+    // Oversample is required as of now as it is the only field in the rescore vector
+    private final float numCandidatesFactor;
+
+    public RescoreVectorBuilder(Float numCandidatesFactor) {
```

Comment:
```suggestion
    public RescoreVectorBuilder(float numCandidatesFactor) {
```

I think we can safely assume this is not an object.

Reply:
Yep, I missed that when changing `numCandidatesFactor` to mandatory. Thanks!